packages/grid/x-grid-modules/src/hooks/root/useKeyboard.ts

```diff
@@ -106,19 +107,25 @@ export const useKeyboard = (options: GridOptions, initialised: boolean, apiRef:
       nextCellIndexes.colIndex =
         nextCellIndexes.colIndex >= colCount ? colCount - 1 : nextCellIndexes.colIndex;
 
-      apiRef.current!.scrollToIndexes(nextCellIndexes);
-      setTimeout(() => {
-        const nextCell = getCellElementFromIndexes(root, nextCellIndexes);
+      apiRef.current!.once(SCROLLING, () => {
+        requestAnimationFrame(() => {
```

Comment:
If we wait for the next animation frame, we might want to can cancel it if we have a new event coming.

Reply:
Hmm yes! Done :)